exp/api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -55,41 +56,223 @@ func (r *AzureManagedMachinePool) Default(client client.Client) {
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type.
 func (r *AzureManagedMachinePool) ValidateCreate(client client.Client) error {
 	azuremanagedmachinepoollog.Info("validate create", "name", r.Name)
+	var allErrs field.ErrorList
+
+	// If AutoScaling is enabled, both MinCount and MaxCount should be set
+	if r.Spec.AutoScaling != nil && (r.Spec.AutoScaling.MaxCount == nil || r.Spec.AutoScaling.MinCount == nil || (*r.Spec.AutoScaling.MaxCount < *r.Spec.AutoScaling.MinCount)) {
+		allErrs = append(allErrs,
```

Comment:
Why are we validating again for MinCount and MaxCount ? 
Kubebuilder already does that right ?
```
	// MaxCount - Maximum number of nodes for auto-scaling
	// +kubebuilder:validation:Required
	MaxCount *int32 `json:"maxCount"`

	// MinCount - Minimum number of nodes for auto-scaling
	// +kubebuilder:validation:Required
	MinCount *int32 `json:"minCount"`

```

Reply:
fixed